File: extensions/youtube/src/main/java/app/morphe/extension/youtube/patches/theme/LogoColorPatch.java
package app.morphe.extension.youtube.patches.theme;

import android.graphics.Color;

import app.morphe.extension.shared.Logger;
import app.morphe.extension.shared.Utils;
import app.morphe.extension.youtube.settings.Settings;

import static app.morphe.extension.shared.StringRef.str;

@SuppressWarnings("unused")
public final class LogoColorPatch {

    private static final boolean LOGO_CUSTOM_COLOR_ENABLED = Settings.LOGO_CUSTOM_COLOR.get();

    // TODO: Add injection points here once bytecode fingerprints are implemented.
}

File: patches/src/main/kotlin/app/morphe/patches/youtube/layout/logocolor/LogoColorPatch.kt
package app.morphe.patches.youtube.layout.logocolor

import app.morphe.patcher.patch.bytecodePatch
import app.morphe.patches.shared.misc.settings.preference.SwitchPreference
import app.morphe.patches.shared.misc.settings.preference.TextPreference
import app.morphe.patches.shared.misc.settings.preference.InputType
import app.morphe.patches.youtube.misc.extension.sharedExtensionPatch
import app.morphe.patches.youtube.misc.settings.PreferenceScreen
import app.morphe.patches.youtube.shared.Constants.COMPATIBILITY_YOUTUBE

private const val EXTENSION_CLASS = "Lapp/morphe/extension/youtube/patches/theme/LogoColorPatch;"

val logoColorPatch = bytecodePatch(
    description = "Set a custom color of the YouTube and Shorts logo inside app",
) {
    dependsOn(
        sharedExtensionPatch,
    )

    compatibleWith(COMPATIBILITY_YOUTUBE)

    execute {
        PreferenceScreen.GENERAL.addPreferences(
            SwitchPreference("morphe_logo_custom_color"),
            TextPreference(
                "morphe_logo_custom_color_value",
                tag = "app.morphe.extension.shared.settings.preference.ColorPickerPreference",
                inputType = InputType.TEXT_CAP_CHARACTERS
            )
        )

        // TODO: Add bytecode hooks here when fingerprints are implemented.
    }
}
